pkg/asset/installconfig/openstack/validation/platform.go

```diff
@@ -57,7 +57,8 @@ func validateMachinesSubnet(p *openstack.Platform, n *types.Networking, ci *Clou
 
 // validateExternalNetwork validates the user's input for the externalNetwork and returns a list of all validation errors
 func validateExternalNetwork(p *openstack.Platform, ci *CloudInfo, fldPath *field.Path) (allErrs field.ErrorList) {
-	if ci.ExternalNetwork == nil {
+	// Return an error if external network was specified in the install config, but hasn't been found
+	if p.ExternalNetwork != "" && ci.ExternalNetwork == nil {
```

Comment:
Are we confident we can always populate `ci.ExternalNetwork`?

Reply:
> Are we confident we can always populate ci.ExternalNetwork?
ci.ExternalNetwork can be nil if either it is not set in the config or no networks with this name were found: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/openstack/validation/cloudinfo.go#L154-L173
In all other cases you will see a value here.

> Is it always possible to list public networks, or is "public network listing" a new required permission?
I assume this is an OpenStack default to allow all users to list public networks (and other public resources). We've always done this to convert a network name into its identifier.

Reply:
OK, perfect! Thank you.